platform/reference/migrations/4-0-migration.mdx

```diff
@@ -0,0 +1,129 @@
+---
+title: Upgrade Guide from pre-v4.0 to v4.0+
+description: How to upgrade from pre-v4.0 to v4.0 release
+sidebar_position: 1
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+These docs cover upgrading from pre v4.0 to v4.0. Once you upgrade to v4.0+, you need to follow the regular [upgrade docs](../../administer/upgrade-migrate/upgrade). 
+
+## Upgrade to the latest Loft v3.x version
+
+Before attempting to perform this major upgrade, upgrade to the latest v3.x version. 
+Review the [older upgrade docs](https://loft.sh/docs/admin/upgrade-loft) on how to upgrade. It is recommended to upgrade using the CLI and not the UI as there are known issues to upgrade versions through the UI.
+
+## Pre-Upgrade Recommendations 
+
+### Read through the v4.0 Release Notes
+
+Review the [release notes for v4.0](https://github.com/loft-sh/loft-enterprise/releases/tag/v4.0.0) to read the updates in the release and understand any behavior changes. 
+
+### vCluster versions are required on virtual clusters and virtual cluster templates
+
+One of the breaking changes in v4.0 is the requirement of a vCluster version defined on your virtual cluster. Previously if you deployed a vCluster version or used a vCluster version without editing the version field, the version was not set in the `spec` of the resource. With each upgrade of the Platform,
+the vCluster would automatically upgrade these virtual clusters to the latest default vCluster version. Due to the [breaking changes introduced in v0.20](https://roadmap.vcluster.com/changelog/vcluster-v020-ga), 
+automatic upgrades of vCluster version are no longer supported and a vCluster version must be set in the spec of a virtual cluster or virtual cluster template. 
+
+We recommend confirmging that the versions of your virtual clusters and virtual cluster templates are in the resource yaml before upgrading. Otherwise,
+your virtual clusters will have errors requiring you to define these versions post upgrade and since it will not know what version you upgraded from, it will not know what
+version to set and manual steps will need to happen to determine the correct version and set it. 
+
+#### How to define the vCluster version
+
+In the UI, type the version of the virtual cluster or virtual cluster template in the textbox and the resource yaml will automatically update the spec to include the version.
```

Comment:
This is during vCluster creation or vCluster template creation. Can we pls add that detail ?

Reply:
It's not during the creation, it's editing as this is nested under the section of how to fix the ones that are missing.

Reply:
But I've added some text around that fact since you thought it was during creation. 

These are all steps to prep your existing deployed virtual clusters on a platform.